### PR TITLE
Fixes issue #917

### PR DESCRIPTION
**Pull request by rabbah**
Do not accept activation id longer than 32 chars if '-' is missing. Add tests.

### CODE REVIEW

**Comment by mdeuser** [comment]
This fix now results in `invalid activation id` being returned from the server.

I noticed other error message strings in the commit that include additional information that indicates more specifically about what is wrong with the id  - i.e.  `activation id is malformed (too long)`,  `activation id is malformed (too short)`.   Was this additional information expected to be sent in the server response?  Would be nice..

**Comment by rabbah** [comment]
yes the message will come back as part of the API response. i can add an additional test for that.

**Comment by mdeuser** [comment]
Should my local test have seen the `activation id is malformed (too long)` error message from the server?  My test run resulted in the `invalid activation id` message being received.

**Comment by rabbah** [comment]
Addressed your comment @mdeuser in https://github.com/openwhisk/openwhisk/pull/918/commits/445e81b8bba387103e11edad277ac275e309c2cf.

**Comment by rabbah** [comment]
IPG 350 approved.

**Comment by rabbah** [comment]
Rebased and squashed.

File: common/scala/src/main/scala/whisk/core/entity/ActivationId.scala [hunk 1]
```diff
@@ -88,13 +91,23 @@ protected[core] object ActivationId extends ArgNormalizer[ActivationId] {
         def read(value: JsValue) = Try {
             val JsString(s) = value
```
Comment (psuter) [inline common/scala/src/main/scala/whisk/core/entity/ActivationId.scala]: Can the id ever contain only numbers, be decoded into `JsNumber`, and have this throw a `MatchError`?

Comment (rabbah) [inline common/scala/src/main/scala/whisk/core/entity/ActivationId.scala]: The schema dictated the activation id is a string. Will add test though and harden for JsNumber. Thanks.

Comment (rabbah) [inline common/scala/src/main/scala/whisk/core/entity/ActivationId.scala]: Addressed in https://github.com/openwhisk/openwhisk/pull/918/commits/c057031eb1661743eb87539d424b97959544b102

File: common/scala/src/main/scala/whisk/core/entity/ActivationId.scala [hunk 2]
```diff
@@ -89,25 +90,44 @@ protected[core] object ActivationId extends ArgNormalizer[ActivationId] {
...
             }
         } match {
             case Success(a)                                 => a
             case Failure(DeserializationException(t, _, _)) => deserializationError(t)
             case Failure(t)                                 => deserializationError("activation id is malformed")
         }
     }
+
+    private def bigIntToActivationId(n: BigInt): ActivationId = {
+        // print the bigint using base 10 than convert to base 16
```
Comment (markusthoemmes) [inline common/scala/src/main/scala/whisk/core/entity/ActivationId.scala]: *then

**Comment by psuter** [comment]
Merged as 8812ced660f568c79ca8511cbae63eadcbb19f95.

**Pull request closed without merging**